docs/source/07_extend_kedro/03_custom_datasets.md

```diff
@@ -133,10 +132,18 @@ class ImageDataSet(AbstractDataSet):
             Data from the image file as a numpy array
         """
         # using get_filepath_str ensures that the protocol and path are appended correctly for different filesystems
-        load_path = get_filepath_str(self._get_load_path(), self._protocol)
+        load_path = get_filepath_str(self._filepath, self._protocol)
         with self._fs.open(load_path) as f:
             image = Image.open(f).convert("RGBA")
             return np.asarray(image)
+
+    def _save(self, data: np.ndarray) -> None:
+        """Saves image data to the specified filepath"""
+        ...
+
+    def _describe(self) -> Dict[str, Any]:
+        """Returns a dict that describes the attributes of the dataset"""
+        ...
```

Comment:
I would actually remove these lines. I believe it was intentional to only focus on one function at a time, rather than include the boilerplate at every stage. The very first step is to define the outline of the dataset, which includes empty `load`, `save` and `describe` methods (L46-L77).

Reply:
Agree with this - I will revert this change.

Reply:
Done in 7b1a46b

Reply:
I understand your intent to focus on the changes and that's good. I do however find it very confusing in this specific example since the code block looks like a complete file. Some of the other code blocks are more obvious that they are only small snippets, but not this one. In the last code snippet, the one on handling credentials, the ellipsis object ("...") is used to indicate that there are more methods in the class. That could be used here too?

Reply:
Fair point, thank you for explaining @daniel-falk. I like your proposal of `...` - seems like a good compromise between focusing on one function at a time and indicating that it's not complete code. I've done this in 7db9b72481664cf3f01ad52eae93cabf295e7229.